### PR TITLE
Add autofix to function-max-empty-lines

### LINKED ISSUE #3613: Add autofix to function-max-empty-lines
> What is the problem you're trying to solve?

Adds autofix of rules [defined in `stylelint-config-standard`](https://github.com/stylelint/stylelint-config-standard/blob/master/index.js) so that everyone can start using [`stylelint-config-standard`](https://github.com/stylelint/stylelint-config-standard) easily.

> What solution would you like to see?

Add autofix to [function-max-empty-lines](https://github.com/stylelint/stylelint/tree/master/lib/rules/function-max-empty-lines)

### PR DESCRIPTION
**Pull request by Bilie**
> Which issue, if any, is this issue related to?

Closes #3613

> Is there anything in the PR that needs further explanation?

No, it's self explanatory

### CODE REVIEW

File: lib/rules/function-max-empty-lines/index.js [hunk 1]
```diff
@@ -53,6 +53,17 @@ const rule = function(max) {
               index -= 1;
             }
 
+            if (context.fix) {
+              decl.value = (
```
Comment (ota-meshi) [inline lib/rules/function-max-empty-lines/index.js]: If value contains a comment, you must use `decl.raws.value.raw`.

File: lib/rules/function-max-empty-lines/index.js [hunk 2]
```diff
@@ -53,6 +53,17 @@ const rule = function(max) {
...
+                decl.toString().substring(0, index) +
+                decl.toString().substring(match.endIndex)
+              )
+                .split(":")[1]
```
Comment (ota-meshi) [inline lib/rules/function-max-empty-lines/index.js]: You can use `declarationValueIndex.js` to take the starting position.

Comment (ota-meshi) [reply lib/rules/function-max-empty-lines/index.js]: Can you add the following test case?

```css
a { transform: /*c*/translate(\n\n\n1,1); }
```

**Comment by ota-meshi** [comment]
Thank you for this PR.
I wrote some change requests in comments.

File: lib/rules/function-max-empty-lines/__tests__/index.js [hunk 3]
```diff
@@ -6,6 +6,7 @@ const { messages, ruleName } = rule;
 testRule(rule, {
   ruleName,
   config: [0],
+  fix: true,
```
Comment (ota-meshi) [inline lib/rules/function-max-empty-lines/__tests__/index.js]: Can you add a test case that autofixe two places at once?

**Comment by Bilie** [comment]
Hi, thank you for reviewing my PR :) I have addressed the comments you left, let me know if there is something that should be improved!

File: lib/rules/function-max-empty-lines/__tests__/index.js [hunk 4]
```diff
@@ -48,50 +49,79 @@ testRule(rule, {
...
+    },
     {
       code: "a { transform: translate(\n\n1\n,\n1\n); }",
+      fixed: "a { transform: translate(\n1\n,\n1\n); }",
+      message: messages.expected(0),
+      line: 1,
+      column: 26
+    },
+    {
+      code: "a { transform: translate(\n\n1\n,\n1\n\n); }",
```
Comment (ota-meshi) [inline lib/rules/function-max-empty-lines/__tests__/index.js]: I should have made it a little clearer.
In the next case, the fixed css becomes invalid.

```css
a { transform: translate(\n\n1\n\n,\n1\n\n); }
```
->
```css
a { transform: translate(\n1\n,\n1\n\n; }
```

**Comment by ota-meshi** [comment]
@Bilie Thank you for the great work :)
I have a few more request.

File: lib/rules/function-max-empty-lines/__tests__/index.js (hunk 4 above)
Comment (Bilie) [reply lib/rules/function-max-empty-lines/__tests__/index.js]: Hi, thank you for the example, it was very helpful! I have updated the PR accordingly

**Comment by ota-meshi** [comment]
@Bilie LGTM! Thank you.

**Comment by ntwb** [comment]
Changelog: https://github.com/stylelint/stylelint/commit/b202c59f1773b67f7a970a374ceb9682bff134d2
> Added: function-max-empty-lines autofix (#3645).

**Pull request merged**